### PR TITLE
Feat(x2a): collectArtifacts with HMAC authz

### PR DESCRIPTION
**Pull request by eloycoto**
### **User description**
- Add a new auth system using HMAC fingerprints.
- Create SignatureValidator utility class for secure signature operations
- Update job tokens to 256-bit
- Take care of time-replay attacks(3 hours)
- Added test for all corner cases.
- Update openapi specs.

I tested with this script:

```

set -e

BASE_URL="${X2A_BASE_URL:-http://localhost:7007/api/x2a}"
AUTH_URL="${AUTH_URL:-http://localhost:7007/api/auth}"
PROJECT_ID="$1"
PHASE="${2:-init}"
DB_PATH="${X2A_DB_PATH:-./packages/backend/x2a.sqlite}"

if ! command -v jq &> /dev/null; then
    echo "Error: jq is required. Install with: sudo dnf install jq"
    exit 1
fi

if ! command -v sqlite3 &> /dev/null; then
    echo "Error: sqlite3 is required. Install with: sudo dnf install sqlite"
    exit 1
fi

echo ""
echo "Creating job..."
echo "==============="

echo "Authenticating with guest token..."

AUTH_RESPONSE=$(curl -s -X GET "$AUTH_URL/guest/refresh" \
  -H "Content-Type: application/json" 2>&1)

TOKEN=$(echo "$AUTH_RESPONSE" | jq -r '.backstageIdentity.token // empty')

if [ -z "$TOKEN" ]; then
    echo "✗ Failed to get authentication token"
    echo "Response: $AUTH_RESPONSE"
    echo ""
    echo "Is the backend running at $AUTH_URL?"
    exit 1
fi

echo "✓ Authenticated"

if [ -z "$PROJECT_ID" ]; then
    echo "Creating new project..."

    PROJECT_RESPONSE=$(curl -s -X POST "$BASE_URL/projects" \
      -H "Content-Type: application/json" \
      -H "Authorization: Bearer $TOKEN" \
      -d '{
        "name": "Test Project '"$(date +%s)"'",
        "description": "Test project",
        "abbreviation": "TP'"$(date +%s | tail -c 4)"'",
        "sourceRepoUrl": "https://github.com/test/source",
        "targetRepoUrl": "https://github.com/test/target",
        "sourceRepoBranch": "main",
        "targetRepoBranch": "main"
      }')

    PROJECT_ID=$(echo "$PROJECT_RESPONSE" | jq -r '.id // empty')

    if [ -z "$PROJECT_ID" ]; then
        echo "✗ Failed to create project"
        echo "Response: $PROJECT_RESPONSE"
        exit 1
    fi

    echo "✓ Project created: $PROJECT_ID"
fi

echo "Creating $PHASE job..."

if [ "$PHASE" != "init" ]; then
    echo "Creating module for $PHASE phase..."

    MODULE_RESPONSE=$(curl -s -X POST "$BASE_URL/projects/$PROJECT_ID/modules" \
      -H "Content-Type: application/json" \
      -H "Authorization: Bearer $TOKEN" \
      -d '{
        "name": "Test Module",
        "sourcePath": "/cookbooks/test"
      }')

    MODULE_ID=$(echo "$MODULE_RESPONSE" | jq -r '.id // empty')

    if [ -z "$MODULE_ID" ]; then
        echo "✗ Failed to create module"
        echo "Response: $MODULE_RESPONSE"
        exit 1
    fi

    echo "✓ Module created: $MODULE_ID"

    # Module-level job
    JOB_RESPONSE=$(curl -s -X POST "$BASE_URL/projects/$PROJECT_ID/modules/$MODULE_ID/run" \
      -H "Content-Type: application/json" \
      -H "Authorization: Bearer $TOKEN" \
      -d '{
        "phase": "'"$PHASE"'",
        "sourceRepoAuth": {"token": "test-token"},
        "targetRepoAuth": {"token": "test-token"}
      }')
else
    # Project-level init job (no phase parameter needed)
    JOB_RESPONSE=$(curl -s -X POST "$BASE_URL/projects/$PROJECT_ID/run" \
      -H "Content-Type: application/json" \
      -H "Authorization: Bearer $TOKEN" \
      -d '{
        "sourceRepoAuth": {"token": "test-token"},
        "targetRepoAuth": {"token": "test-token"}
      }')
fi

JOB_ID=$(echo "$JOB_RESPONSE" | jq -r '.jobId // empty')

if [ -z "$JOB_ID" ] || [ "$JOB_ID" = "null" ]; then
    echo "✗ Failed to create job"
    echo "Response: $JOB_RESPONSE"
    exit 1
fi

echo "✓ Job created: $JOB_ID"

echo "Fetching callback token from database..."

if [ ! -f "$DB_PATH" ]; then
    echo "✗ Database file not found at: $DB_PATH"
    echo ""
    echo "Make sure the backend is configured to use a file-based SQLite database."
    echo "Check app-config.yaml: backend.database.connection should be a file path."
    exit 1
fi

CALLBACK_TOKEN=$(sqlite3 "$DB_PATH" "SELECT callback_token FROM jobs WHERE id = '$JOB_ID';" 2>&1)

if [ -z "$CALLBACK_TOKEN" ]; then
    echo "✗ Failed to retrieve callback token"
    echo ""
    echo "Query the database manually:"
    echo "  sqlite3 $DB_PATH \"SELECT id, phase, callback_token FROM jobs WHERE id = '$JOB_ID';\""
    exit 1
fi

echo "✓ Token retrieved"
echo ""
echo "=================================="
echo "Job Created Successfully!"
echo "=================================="
echo ""
echo "Project ID:     $PROJECT_ID"
echo "Job ID:         $JOB_ID"
echo "Phase:          $PHASE"
echo "Callback Token: $CALLBACK_TOKEN"
echo ""
echo "Test with UV (X2Ansible convertor CLI):"
echo "----------------------------------------"
echo "uv run app.py report \\"
echo "    --url \"http://localhost:7007/api/x2a/projects/$PROJECT_ID/collectArtifacts?phase=$PHASE\" \\"
echo "    --job-id \"$JOB_ID\" \\"
echo "    --callback-token \"$CALLBACK_TOKEN\" \\"
echo "    --artifacts \"migration_plan:https://storage.example/plan.md\""
echo ""

echo "" | sed "s/JOB_ID/$JOB_ID/g; s/CALLBACK_TOKEN/$CALLBACK_TOKEN/g; s/PROJECT_ID/$PROJECT_ID/g; s/PHASE/$PHASE/g"
```

Fix FLPATH-3341

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/redhat-developer/rhdh-plugins/blob/main/CONTRIBUTING.md#creating-changesets))
- [ ] Added or Updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

___

### **PR Type**
Enhancement, Tests

___

### **Description**
- Implement HMAC-SHA256 signature validation for collectArtifacts endpoint

- Add SignatureValidator utility class for secure signature operations

- Upgrade callback tokens to 256-bit cryptographic strength

- Add replay attack prevention with 3-hour job age validation

- Refactor request handling with dedicated validation classes

- Update OpenAPI specs with security scheme documentation

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["Job Creation"] -->|"Generate 256-bit token"| B["callbackToken"]
  B -->|"Stored in Job"| C["Job Record"]
  D["X2Ansible Job"] -->|"Sign with token"| E["SignatureValidator"]
  E -->|"HMAC-SHA256"| F["X-Callback-Signature"]
  F -->|"Include in request"| G["collectArtifacts Endpoint"]
  G -->|"Validate signature"| H["AuthenticationHandler"]
  H -->|"Check job age"| I["Replay Prevention"]
  I -->|"3 hour window"| J["Accept/Reject"]
```

<details><summary><h3>File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>5 files</summary><table>
<tr>
  <td><strong>SignatureValidator.ts</strong><dd><code>Create HMAC-SHA256 signature validator utility class</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/redhat-developer/rhdh-plugins/pull/2403/files#diff-2e1bef1c9339ab9475a882a12a788540fc26b1f4190f325cf7b78acb142d46c8">+66/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>collectArtifacts.ts</strong><dd><code>Implement HMAC signature validation and request handling</code>&nbsp; </dd></td>
  <td><a href="https://github.com/redhat-developer/rhdh-plugins/pull/2403/files#diff-0e9181db06383c67343e6efa97bef76d30874ed47db2ed20d0cdf613b277e58c">+277/-81</a></td>

</tr>

<tr>
  <td><strong>projects.ts</strong><dd><code>Upgrade callback token generation to 256-bit</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/redhat-developer/rhdh-plugins/pull/2403/files#diff-272124295e6cd44ec427ee0a337431ee937722b3d2dc1db3706a515107c920bf">+3/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>modules.ts</strong><dd><code>Upgrade callback token generation to 256-bit</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/redhat-developer/rhdh-plugins/pull/2403/files#diff-a178e2ea3dfa24ee48b2ec451aeffaea747f14c9419b24080f59750331688f18">+3/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>index.ts</strong><dd><code>Restructure router to bypass OpenAPI parser for raw body access</code></dd></td>
  <td><a href="https://github.com/redhat-developer/rhdh-plugins/pull/2403/files#diff-70e9659163ac9392b9af2ab4e65cd40008277c3d90ebf085508a991d38d59159">+16/-6</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>SignatureValidator.test.ts</strong><dd><code>Add comprehensive tests for signature validation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/redhat-developer/rhdh-plugins/pull/2403/files#diff-f1ce007985288ece2ea6e2e4cae0765549babdc9be55f60ca4060c0087f8525e">+218/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>collectArtifacts.test.ts</strong><dd><code>Add signature validation test cases and update existing tests</code></dd></td>
  <td><a href="https://github.com/redhat-developer/rhdh-plugins/pull/2403/files#diff-e5878ac1ecbfa53c6f8bbd723734700548ad79d33fa273442b94d76fee567697">+344/-45</a></td>

</tr>
</table></details></td></tr><tr><td><strong>Documentation</strong></td><td><details><summary>5 files</summary><table>
<tr>
  <td><strong>openapi.yaml</strong><dd><code>Document HMAC security scheme and replay attack prevention</code></dd></td>
  <td><a href="https://github.com/redhat-developer/rhdh-plugins/pull/2403/files#diff-150d767ae6339b5797ebd1c65fd10812209d0eca8b619ad069793019e001dd38">+29/-3</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>router.ts</strong><dd><code>Update generated OpenAPI spec with security definitions</code>&nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/redhat-developer/rhdh-plugins/pull/2403/files#diff-e161addc1973f908f1bb24ffd5739d29c3b96ddb11c02c43b2ce9aac53950184">+26/-2</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Api.server.ts</strong><dd><code>Add X-Callback-Signature header to server API type</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/redhat-developer/rhdh-plugins/pull/2403/files#diff-2d850d3e10fcbb15e4708b1e6b28dccb82c0c730e1616fe6fd65cffc36ff1bbf">+4/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Api.client.ts</strong><dd><code>Add X-Callback-Signature header to client API type</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/redhat-developer/rhdh-plugins/pull/2403/files#diff-8516446b0aba82de32aa1960b3fbd249440758b2a6b1aa0c0398c3f21e47edfe">+6/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>report.api.md</strong><dd><code>Update API report with signature header definition</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/redhat-developer/rhdh-plugins/pull/2403/files#diff-036be3ce215d886ce29248791dca69c168e14feb5444763640aa611fe49f0b38">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tbody></table>

</details>

___

### CODE REVIEW

File: workspaces/x2a/plugins/x2a-backend/src/router/collectArtifacts.ts [hunk 1]
```diff
@@ -117,90 +282,121 @@ export function registerCollectArtifactsRoutes(
...
-    throw error;
   }
 
-  if (validatedBody.status === 'error' && !validatedBody.errorDetails) {
-    throw new InputError('errorDetails field is required when status is Error');
-  }
+  const logs = await fetchJobLogs(
+    kubeService,
+    logger,
+    job.k8sJobName as string,
```
Comment (elai-shalev) [inline workspaces/x2a/plugins/x2a-backend/src/router/collectArtifacts.ts]: no need to cast 'as string' as the fetchJobLogs can handle null values gracefully
